doubly_linked_list_test.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+
+from doubly_linked_list import doubly_LinkedList
+
+def test_doubly_linked_list_init():
+    my_list = doubly_LinkedList([1, 2, 3])
+    assert my_list._size == 3
+    assert my_list.head.prev == None
+    assert my_list.tail.value == 1
+    assert my_list.tail.next == None
+
+def test_doubly_linked_list_init_empty():
+    my_list = doubly_LinkedList()
+    assert my_list._size == 0
+    assert my_list.head == None
+    assert my_list.tail == None
+
+def test_doubly_linked_list_append():
+    my_list = doubly_LinkedList([1, 2])
+    my_list.append(3)
+    assert my_list.tail.value == 3
+    assert my_list.tail.next.value == 1
+    assert my_list._size == 3
+
+def test_doubly_linked_list_shift():
+    my_list = doubly_LinkedList([1, 2])
+    assert my_list.shift() == 1
+    assert my_list.tail.value == 2
+    assert my_list.tail.next == None
+    assert my_list._size == 1
+
+def test_doubly_linked_list_shift_empty():
+     with pytest.raises(ValueError, match="Shift does not work on an empty list"):
+        my_list = doubly_LinkedList()
+        my_list.shift()
+
+def test_doubly_linked_list_push_empty():
+    my_list = doubly_LinkedList([1, 2])
+    my_list.push('')
+    assert my_list._size == 3
+    assert my_list.head.value == ''
+
+def test_doubly_linked_list_push():
```

Comment:
I want to see all methods that manipulate the list, `push`, `pop`, `append`, `shift` to start at 0 or 3 long and go all the way the other direction with the structure comprehensively checked on each step.

Example:
```
dll = DoublyLinkedList([3,2,1])
assert dll.head.prev = None
assert dll.head.value = 1
assert dll.head.next.value = 2
assert dll.head.next.next.value = 3
assert dll.head.next.next.next = None
assert dll.tail.next = None
assert dll.tail.value = 3
assert dll.tail.prev.value = 2
assert dll.tail.prev.prev.value = 1
assert dll.tail.prev.prev.prev = None
assert dll._size = 3

assert dll.pop() == 1
assert dll.head.prev = None
assert dll.head.value = 2
assert dll.head.next.value = 3
assert dll.head.next.next = None
assert dll.tail.next = None
assert dll.tail.value = 3
assert dll.tail.prev.value = 2
assert dll.tail.prev.prev = None
assert dll._size = 2

assert dll.pop() == 2
assert dll.head.prev = None
assert dll.head.value = 3
assert dll.head.next = None
assert dll.tail.next = None
assert dll.tail.value = 3
assert dll.tail.prev = None
assert dll._size = 1

assert dll.pop() == 3
assert dll.head = None
assert dll.tail = None
assert dll._size = 0
```

I know this seems like a lot, but you'll see why it's import as you try to get the tests to pass.